public/components/Flyout/flyouts/createDetector.js

```diff
@@ -0,0 +1,814 @@
+import { MAX_INTERVAl_LENGTH_MINUTES } from '../../../pages/MonitorDetails/containers/utils/helpers';
+
+function toString(obj) {
+  if (typeof obj != 'undefined') {
+    if (obj.hasOwnProperty('period')) {
+      let period = obj.period;
+      return period.interval;
+    } else if (typeof obj == 'number') {
+      // epoch
+      return moment(obj).format('MM/DD/YY hh:mm A');
+    }
+  }
+  return '-';
+}
+
+class FeaturePreview extends Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    const featureAttributes = this.props.featureAttributes;
+    const items = [
+      {
+        name: featureAttributes.feature_name,
+        definition: '',
+        state: 'Enabled',
+      },
+    ];
+    const columns = [
+      {
+        field: 'name',
+        name: 'Feature name',
+      },
+      {
+        field: 'definition',
+        name: 'Feature definition',
+        render: () => {
+          return (
+            <div>
+              <p>
+                {' '}
+                <b>Field:</b> {featureAttributes.fieldName || ''}
+              </p>
+              <p>
+                {' '}
+                <b>Aggregation method:</b> {featureAttributes.aggregationType || ''}
+              </p>
+            </div>
+          );
+        },
+      },
+      {
+        field: 'state',
+        name: 'State',
+      },
+    ];
+    return (
+      <EuiFlexGroup direction="column" gutterSize="s" style={{ paddingTop: '20px' }}>
+        <EuiFlexItem>
+          <EuiTitle size="s">
+            <h3>Features</h3>
+          </EuiTitle>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiPanel grow={false}>
+            <EuiBasicTable
+              items={items}
+              columns={columns}
+              cellProps={() => {
+                return {
+                  textOnly: true,
+                };
+              }}
+            />
+          </EuiPanel>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+}
+
+export class FilterDisplay extends Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    let filter = this.props;
+    if (filter === 'all fields are included') {
+      return (
+        <EuiText>
+          <p className="enabled">-</p>
+        </EuiText>
+      );
+    } else {
+      return <EuiText>{filter}</EuiText>;
+    }
+  }
+}
+const FixedWidthRow = (props) => <EuiFormRow {...props} style={{ width: '150px' }} />;
+
+async function createAndStartDetector(context) {
+  const configs = context.adConfigs;
+  const httpClient = context.httpClient;
+  try {
+    const response = await httpClient.post('../api/alerting/detectors', {
+      configs,
+    });
+    const {
+      data: {
+        ok,
+        response: { _id },
+      },
+    } = response;
+    const detectorId = _id;
+    if (ok) {
+      try {
+        const response = await httpClient.post(`../api/alerting/detectors/${detectorId}/_start`);
```

Comment:
Should we ask user to confirm should start detector or not after creation ?

Reply:
The general idea is to make the creation process as user friendly as possible so giving an option not to start the detector might confuse unexperienced users. Overall this follows the "1-click" design better and especially assuming the data is already being used for a monitor it is likely they would just want to start this. However I see the point in the added confirmation and wait for the actual start, I'll mention it with UX and add to the handoff docs.